### PR TITLE
Fix buildQueryString params types

### PR DESCRIPTION
**Pull request by npelletm**
* fix buildQueryString parameters types
* add typing on private methods
* various var chore

### CODE REVIEW

**Comment by EisenbergEffect** [comment]
This PR is a bit hard to follow. It looks like the spacing changed a lot. Can you possible ensure that you are using the same 2 spaces rule as us? Also, if you can void including the dist folder in your commit, that would also help. Finally, if possible, when you have multiple commits, would you mind squashing them?

I'm going to close them for now. If you can resubmit as above, that would greatly help.

**Comment by npelletm** [comment]
Sorry, I am a beginner on github PR.
I review my commit with your advices and resubmit another PR.
